src/main/java/application/model/OrderItem.java

```diff
@@ -0,0 +1,45 @@
+package application.model;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.Table;
+import java.math.BigDecimal;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.NonNull;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
+
+@Entity
+@Data
+@NoArgsConstructor
+@SQLDelete(sql = "UPDATE order_items SET is_deleted = true")
+@Where(clause = "is_deleted = false")
+@Table(name = "order_items")
+public class OrderItem {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @ManyToOne
+    @JoinColumn(name = "order_id")
+    @NonNull
+    private Order order;

```

Comment:
Don't forget to exclude relations from equals and hash code

Reply:
I guess still a valid point to consider